src/App.tsx

```diff
@@ -1,97 +1,94 @@
-  const { name, born, died } = peopleFromServer[0];
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [selectedPeople, setSelectedPeople] = useState<Person | null>(null);
+  const [showList, setShowList] = useState(false);
+  const [pageTitle, setPageTitle] = useState<string>('No selected person');
 
-  return (
-    <div className="container">
-      <main className="section is-flex is-flex-direction-column">
-        <h1 className="title" data-cy="title">
-          {`${name} (${born} - ${died})`}
-        </h1>
-
-        <div className="dropdown is-active">
-          <div className="dropdown-trigger">
-            <input
-              type="text"
-              placeholder="Enter a part of the name"
-              className="input"
-              data-cy="search-input"
-            />
-          </div>
+  const applyQuery = useCallback(
+    debounce((inputQuery: string) => setAppliedQuery(inputQuery), 1000),
+    [setAppliedQuery],
+  );
 
-          <div
-            className="dropdown-menu"
-            role="menu"
-            data-cy="suggestions-list"
-          >
-            <div className="dropdown-content">
-              <div
-                className="dropdown-item"
-                data-cy="suggestion-item"
-              >
-                <p className="has-text-link">Pieter Haverbeke</p>
-              </div>
+  const handleQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const inputQuery = event.target.value;
 
-              <div
-                className="dropdown-item"
-                data-cy="suggestion-item"
-              >
-                <p className="has-text-link">Pieter Bernard Haverbeke</p>
-              </div>
+    setQuery(inputQuery);
+    applyQuery(inputQuery);
+  };
 
-              <div
-                className="dropdown-item"
-                data-cy="suggestion-item"
-              >
-                <p className="has-text-link">Pieter Antone Haverbeke</p>
-              </div>
+  useEffect(() => {
+    if (!query) {
+      setSelectedPeople(null);
+    }
+  }, [query]);
 
-              <div
-                className="dropdown-item"
-                data-cy="suggestion-item"
-              >
-                <p className="has-text-danger">Elisabeth Haverbeke</p>
-              </div>
+  useEffect(() => {
+    setPageTitle(() => {
+      return selectedPeople
+        ? `${selectedPeople.name} (${selectedPeople.born} - ${selectedPeople.died})`
+        : 'No selected person';
+    });
+  }, [selectedPeople]);
 
-              <div
-                className="dropdown-item"
-                data-cy="suggestion-item"
-              >
-                <p className="has-text-link">Pieter de Decker</p>
-              </div>
+  const filteredPeople = useMemo(() => {
+    return peopleFromServer.filter((person) => person.name
+      .toLowerCase().includes(appliedQuery.trim().toLowerCase()));
```

Comment:
`person.name.toLowerCase().includes(appliedQuery.trim().toLowerCase())` separate this to different variables to improve readability and codestyle. It will be a good practice